moped-editor/src/views/projects/projectView/ProjectComponents/index.js

```diff
@@ -269,43 +173,76 @@ export default function MapView({ projectName, projectStatuses }) {
           <PlaceholderToolbar />
           <div className={classes.drawerContainer}>
             <List>
-              {!isEditingComponent && (
-                <>
-                  <ListItem dense>
-                    <Button
-                      size="small"
-                      color="primary"
-                      fullWidth
-                      startIcon={<AddCircleOutlineIcon />}
-                      onClick={onStartCreatingComponent}
-                    >
-                      New Component
-                    </Button>
-                  </ListItem>
-                  <Divider />
-                </>
-              )}
-              {draftComponent && (
+              {!createState.isCreatingComponent &&
+                !editState.isEditingComponent && (
+                  <>
+                    <ListItem dense>
+                      <Button
+                        size="small"
+                        color="primary"
+                        fullWidth
+                        startIcon={<AddCircleOutlineIcon />}
+                        onClick={onStartCreatingComponent}
+                      >
+                        New Component
+                      </Button>
+                    </ListItem>
+                    <Divider />
+                  </>
+                )}
+              {createState.draftComponent &&
+                createState.isCreatingComponent && (
+                  <DraftComponentListItem
+                    primaryText={createState.draftComponent.component_name}
+                    secondaryText={createState.draftComponent.component_subtype}
+                    onSave={onSaveDraftComponent}
+                    onCancel={onCancelComponentCreate}
+                    saveButtonDisabled={
+                      !createState.draftComponent?.features.length > 0
+                    }
+                    saveButtonText="Save"
+                  />
+                )}
+              {editState.isEditingComponent && (
                 <DraftComponentListItem
-                  component={draftComponent}
-                  onSave={onSaveComponent}
-                  onCancel={onCancelComponentCreate}
+                  primaryText={
+                    clickedComponent?.moped_components?.component_name ||
+                    editState.draftEditComponent?.moped_components
+                      ?.component_name
+                  }
+                  secondaryText={
+                    clickedComponent?.moped_components?.component_subtype ||
+                    editState.draftEditComponent?.moped_components
+                      ?.component_subtype
+                  }
```

Comment:
I'm not feeling great about switching between `clickedComponent` and `draftEditComponent` here. They are entangled because the UI shows an expanded list item on the left when a `clickedComponent` is set. 

The issue comes up when `clickedComponent` is cleared when a user clicks anywhere on the map to the expanded list item collapses (to no longer highlight it). We want that to happen, but, if they are editing, we lose focus on the item being edited and the component attributes go blank in the list item.

When editing starts, I copy `clickedComponent` state into `draftEditComponent` so we retain the attributes and features throughout the editing flow. I've thought a lot about this and am stuck on finding another way.

Reply:
Alright, I thought about this some more, and the issue is that I should have copied `clickedComponent` to the `draftEditComponent` state right when the "Edit" button is clicked (instead of after the "Map" button in the edit modal is clicked).

Also changed the other list items to conditionally render when not editing or creating. This takes away the ability to even see the other components and their zoom buttons. We can take another look down the line if we want different behavior, but hoping this keeps things simple for now.